ext/systemd-daemon/systemd-daemon.c

```diff
@@ -16,6 +16,54 @@ static VALUE _sd_notify(VALUE mod, VALUE unset_env, VALUE state)
   return INT2FIX(return_code);
 }
 
+/* Supports only one fd per call */
+static VALUE _sd_pid_notify_with_fds(VALUE mod, VALUE pid, VALUE unset_env, VALUE state, VALUE fds, VALUE fds_size)
+{
+  int i, return_code;
+
+#ifdef HAVE_SD_PID_NOTIFY_WITH_FDS
+  const char *sd_state;
+  int size = FIX2INT(fds_size);
+  int *fds_ref = (int *)malloc(size * sizeof(int));
+
```

Comment:
Since we have the https://github.com/ctrochalakis/systemd-daemon/pull/3/files#diff-c16eb40418a86944f87713e75287c5f5R14, aren't we sure that this will be an array?